packages/app/src/features/actions/flavours/stake/logic/createStakeActions.ts

```diff
@@ -65,10 +65,10 @@ export function createStakeActions(objective: StakeObjective, context: ActionCon
         mode: 'withdraw',
       }
 
-      const stakeAmount =
-        objective.token.symbol === sDaiSymbol
-          ? savingsDaiInfo.convertToAssets({ shares: objective.amount })
-          : savingsUsdsInfo.convertToAssets({ shares: objective.amount })
+      const withdrawReceipt = context.txReceipts.find(([action]) => action.type === 'withdrawFromSavings')?.[1]
```

Comment:
Let's extract findOne helper function and use destruction here.

```suggestion
      const [_, withdrawReceipt] = findOne(context.txReceipts, ([action]) => action.type === 'withdrawFromSavings')````

Reply:
Can't be done as `withdrawReceipt` can actually be undefined